### PR TITLE
Changed TimeEntry.StartedTime to a string

### PR DESCRIPTION
**Pull request by jamesburns-rts**
Hi, thanks for making this library!

I ran into an issue whenever getting my entries whenever I had an active timer going. In the response, the entry had the field `"started_time": "2:24pm"`, which could not automatically be unmarshalled into `time.TIme` and it threw an error.

I just swapped the field type out for a `*string` for a quick workaround, but you may have other ideas of what to do.

### CODE REVIEW

File: harvest/time.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	time.Time
+}
+
+func (t Time) String() string {
+	return strings.ToLower(t.Time.Format(time.Kitchen))
+}
+
+func (t *Time) UnmarshalJSON(data []byte) (err error) {
+	str := string(data)
+	str = strings.Trim(str, "\"")
```
Comment (becoded) [inline harvest/time.go]: REQ: It seems this isn't covering every case.
When doing some tests with your changes, I received `null` and `23:57` instead

File: harvest/time.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	str = strings.ToUpper(str)
+	resp, err := time.Parse(time.Kitchen, str)
+	if err != nil {
+		return err
+	}
+
+	now := time.Now()
+	(*t).Time = time.Date(now.Year(), now.Month(), now.Day(), resp.Hour(), resp.Minute(), 0, 0, now.Location())
```
Comment (becoded) [inline harvest/time.go]: Q: have you tested this when you selected entries from last month? Do you get the startedTime back from the API?
If so, your date will be wrongly set to the current date.

**Comment by becoded** [comment]
Thanks @jamesburns-rts for the bugfix and extra methods you added.
I had a quick look at it and found some issues with your code.
Could you have a look at it?

**Comment by jamesburns-rts** [comment]
I didn't think about a timer that had been started before the current day, so I just switched them back to strings since the same information is in `TimerStartedAt`. Let me me know if you think there is a better way to handle this.

**Comment by becoded** [comment]
I prefer your previous solution, it just needs some hardening.
It gives you a lot more options to work with if the fields are Time instances instead of a string.
I wouldn't set the date part of the time object. I know it isn't 100% correct but I feel it is acceptable.

You can tackle the different formats in multiple ways
one of them is just parsing first with time.Kitchen and on error parse again with layout `15:04`

I ran https://gist.github.com/smagch/d2a55c60bbd76930c79f and 
`01:40` got converted to `Time:0000-01-01 01:40:00 +0000 UTC`

**Comment by jamesburns-rts** [comment]
Ok, I reverted my change, hardened the parsing, and added some tests. I also changed it to assume the time was in the `time.Local` time zone since it is in EDT for me (I changed this in `Date` as well).

**Comment by becoded** [comment]
Thx for the update and improvements!

**Pull request merged**